### ISSUE TITLE
Spring security is pulling different versions

### ISSUE THREAD

**Issue by mrioan** [stormpath/stormpath-sdk-java#1253]
I am seeing that our Spring Security integration is pulling (see the different versions):

```
org.springframework.security:spring-security-web:4.2.1.RELEASE -> 4.1.4.RELEASE
org.springframework.security:spring-security-config:4.2.1.RELEASE -> 4.1.4.RELEASE
org.springframework.security:spring-security-core:4.1.4.RELEASE
```

In our examples I see that the dependencies are just fine. But in the ticket number 10950 I am seeing this strange behavior (they are using Gradle)

This causes Spring Security to work incorrectly (for example the redirection to login does not take place when an restricted endpoint is trying to be accessed by an unauthenticated user: the access is simply automatically denied)

BTW, they are using IDSIte and Java SDK 1.4.0

**Comment by mrioan** [comment]
This issue is not resolvable from our own end. The problem is that Spring Boot is automatically pulling those old Spring Security versions. This issue is supposed to be automatically fixed when Spring Boot 1.5.0 is released. In the meantime the workaround is to manually add the required Spring Security libraries with version 4.2.1.

In order to fix this issue we will need to update our Java SDK to Spring Boot 1.5.0. So I will keep this issue open until we actually can do that.

**Comment by mraible** [comment]
I believe it's possible to add a single property if you're extending from Spring Boot's parent.

```xml
<spring.security.version>4.2.1</spring.security.version>
```
